packages/autorest.python/autorest/codegen/_utils.py

```diff
@@ -10,3 +10,7 @@
     "Code generated by Microsoft (R) Python Code Generator.\n"
     "Changes may cause incorrect behavior and will be lost if the code is regenerated."
 )
+
+SWAGGER_PACKAGE_MODE = ["mgmtplane", "dataplane"]  # for backward compatibility
+TYPESPEC_PACKAGE_MODE = ["azure-mgmt", "azure-dataplane", "generic"]
+VALID_PACKAGE_MODE = SWAGGER_PACKAGE_MODE + TYPESPEC_PACKAGE_MODE
```

Comment:
This is almost good to go, I like how you made swagger `--packaging-mode` also accept `azure-mgmt` etc. We should verify though that from tsp you can only pass in `TYPESPEC_PACKAGE_MODE` and fail if you pass in anything from `SWAGGER_PACKAGE_MODE`

Reply:
updated